File: FrontEndNext/frontendnext/src/lib/types/review.ts
export interface Review {
  id: number;
  productId: number;
  accountId: number;
  accountName: string;
  rating: number;
  comment: string;
  isApproved?: boolean | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface ReviewCreate {
  productId: number;
  accountId: number;
  rating: number;
  comment: string;
}

export interface ReviewCreateDto {
  productId: number;
  accountId: number;
  rating: number;
  comment: string;
}

export interface ReviewUpdateDto {
  productId: number;
  accountId: number;
  rating: number;
  comment: string;
  isApproved?: boolean;
}

File: BackEnd.Net/src/Alcohol/Models/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Alcohol.Models;

public class Review
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int CustomerId { get; set; }

    [Required]
    public int ProductId { get; set; }

    [Required]
    [Range(1, 5)]
    public int Rating { get; set; }

    [Required]
    [StringLength(1000)]
    public string Comment { get; set; }

    public bool? IsApproved { get; set; } // null = pending, true = approved, false = rejected

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    [ForeignKey("CustomerId")]
    public virtual Account Customer { get; set; }

    [ForeignKey("ProductId")]
    public virtual Product Product { get; set; }
} 